java/src/com.zeroc.icegrid/build.gradle

```diff
@@ -4,9 +4,11 @@ project.ext.displayName = "IceGrid"
 project.ext.moduleName = "com.zeroc.icegrid"
 project.ext.description = "Locate, deploy, and manage Ice servers"
 
-slice {
```

Comment:
It's not clear to me why this syntax change is needed. Will this affect build.gradle written by users?

Reply:
> It's not clear to me why this syntax change is needed.

Is not strictly need, is part of updating the plugin to use a more modern Gradle API.

> Will this affect build.gradle written by users?

Yes. The new plugin uses a Gradle source set for configuring Slice files. The old plugin uses a custom object (Java) https://github.com/zeroc-ice/ice-builder-gradle/blob/main/src/main/groovy/com/zeroc/gradle/icebuilder/slice/Java.groovy#L5-L15

Reply:
So, if I use the `slice {` syntax with Ice 3.7, do I need to change this syntax when I upgrade to 3.8, or this change is recommended but optional - the old syntax still works?

Reply:
The new plugin uses the new syntax. The old plugin would still work there is not a hard pressure for user to upgrade.